### ISSUE TITLE
Is config.rank <= 0 ever true?

### ISSUE THREAD

**Issue by def-roth** [EricLBuehler/candle-lora#19]
I can't see any case where the rank can be negative for Tensor multiplication as for 0 that would be equal to drop everything before.  Please correct me.

**Comment by EricLBuehler** [comment]
@def-roth yes the rank should never be negative. We should probably use `NonZeroUsize`.
